src/cmd/build.rs

```diff
@@ -335,61 +333,47 @@ fn exec_cargo_for_wasm_target(
 fn exec_cargo_dylint(crate_metadata: &CrateMetadata, verbosity: Verbosity) -> Result<()> {
     check_dylint_requirements(crate_metadata.manifest_path.directory())?;
 
-    let tmp_dir = tempfile::Builder::new()
-        .prefix("cargo-contract-dylint_")
-        .tempdir()?;
-    tracing::debug!("Using temp workspace at '{}'", tmp_dir.path().display());
-
-    let driver = include_bytes!(concat!(env!("OUT_DIR"), "/ink-dylint-driver.zip"));
-    crate::util::unzip(driver, tmp_dir.path().to_path_buf(), None)?;
+    let verbosity = if verbosity == Verbosity::Verbose {
+        // `dylint` is verbose by default, it doesn't have a `--verbose` argument,
+        Verbosity::Default
+    } else {
+        verbosity
+    };
 
-    let manifest_path = crate_metadata.manifest_path.cargo_arg()?;
-    let args = vec!["--lib", "ink_linting", &manifest_path];
-    let tmp_dir_path = tmp_dir.path().as_os_str().to_string_lossy();
+    let target_dir = &crate_metadata.target_directory.to_string_lossy();
+    let args = vec!["--lib=ink_linting", "--quiet"];
     let env = vec![
-        ("DYLINT_LIBRARY_PATH", Some(tmp_dir_path.as_ref())),
-        // For tests we need to set the `DYLINT_DRIVER_PATH` to a tmp folder,
-        // otherwise tests running in parallel will try to write to the same
-        // file at the same time which will result in a `Text file busy` error.
-        #[cfg(test)]
-        ("DYLINT_DRIVER_PATH", Some(tmp_dir_path.as_ref())),
-        // We need to remove the `CARGO_TARGET_DIR` environment variable in
+        // We need to set the `CARGO_TARGET_DIR` environment variable in
         // case `cargo dylint` is invoked.
         //
-        // This is because the ink! dylint driver crate found in `dylint` uses a
-        // fixed Rust toolchain via the `ink_linting/rust-toolchain` file. By
-        // removing this env variable we avoid issues with different Rust toolchains
-        // interfering with each other.
-        ("CARGO_TARGET_DIR", None),
+        // This is because we build from a temporary directory (to patch the manifest) but still
+        // want the output to live at a fixed path. `cargo dylint` does not accept this information
+        // on the command line.
+        ("CARGO_TARGET_DIR", Some(target_dir.as_ref())),
```

Comment:
Brilliant this is what I was missing when I looked at doing this!

![image](https://user-images.githubusercontent.com/75586/186199538-69121e01-6498-4f55-b4d1-28bd8631f04d.png)